tiledb/sm/array/array.cc

```diff
@@ -553,32 +553,55 @@ void Array::delete_array(const URI& uri) {
   throw_if_not_ok(this->close());
 }
 
-Status Array::delete_fragments(
+void Array::delete_fragments(
     const URI& uri, uint64_t timestamp_start, uint64_t timestamp_end) {
   // Check that query type is MODIFY_EXCLUSIVE
   if (query_type_ != QueryType::MODIFY_EXCLUSIVE) {
-    return LOG_STATUS(Status_ArrayError(
+    throw StatusException(Status_ArrayError(
```

Comment:
Make a class for these. See SparseGlobalOrderReaderStatusException for an example.